crates/hir-def/src/hir.rs

```diff
@@ -526,7 +526,7 @@ pub enum Pat {
     Tuple { args: Box<[PatId]>, ellipsis: Option<usize> },
     Or(Box<[PatId]>),
     Record { path: Option<Box<Path>>, args: Box<[RecordFieldPat]>, ellipsis: bool },
-    Range { start: ExprId, end: ExprId },
+    Range { start: Option<Literal>, end: Option<Literal> },
```

Comment:
This doesn't seem right, there can be other things than Literals here, see https://github.com/rust-lang/rust-analyzer/issues/12210

Reply:
Oh this should handle constants as well? I think I still would prefer an enum `LiteralOrConst` over using `ExprId` there.

Reply:
I went with `LiteralOrConst` but now I see why we may want to use `ExprId` or `PatId` here (I still not fully sure that `LiteralOrConst` is dead end). Doing range hir lowering correctly is out of scope for this PR I think.

Reply:
>  Doing range hir lowering correctly is out of scope for this PR I think.

Leaving that out is fine yes, I mainly brought it up since oyu removed the FIXME, but I want to have a FIXME back that this isn't correctly handling the entirety yet